### ISSUE TITLE
[Bug]: delete action failed - cannot delete or move a file

### ISSUE THREAD

**Issue by maximelehericy** [nextcloud/server#44631]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I cannot delete a file I recently created on c.nc.c

### Steps to reproduce

1. go to your documents
2. create a file
3. click on delete file
4. file stays, and error popup shows delete action failed

### Expected behavior

File should get deleted

### Installation method

None

### Nextcloud Server version

29

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by maximelehericy** [comment]
cannot move either (could not move <filename>)...

**Comment by sorbaugh** [comment]
> cannot move either (could not move )...

Thanks for reporting, adjusted title!

**Comment by maximelehericy** [comment]
hmmm, I am noticing that after a refresh, the action is taken into account.
A similar behavior can be observed there: https://github.com/nextcloud/server/issues/44637

**Comment by maximelehericy** [comment]
another bug I have when trying to delete several files at once:
![image](https://github.com/nextcloud/server/assets/87175513/122d1ced-8f13-44ed-b827-1e02fda8f42b)

An endless task. In that case, at the end and after a refresh, Files are not deleted.

**Comment by Pytal** [comment]
Does not reproduce locally but does on c.nc 🤔

MOVE returns a `423 Locked` and DELETE never returns, but have no knowledge on file locking

Maybe you'd have a better idea here @juliushaertl ?

**Comment by alperozturk96** [comment]
I have same problem in Android app. I think it's a backend bug.

### How to Reproduce

1. Try to delete a media file in Preview
2. UNHANDLED_HTTP_CODE | 423 will be trowed by backend
3. Try the same operation again a couple of times, then the file will be deleted.

**Comment by artonge** [comment]
I remember @come-nc having a similar issue in Cypress, the issue was that the file was locked during the preview generation, so deleting it failed if the request was made right after its creation. Maybe waiting for the preview request to be done would help.

**Comment by Pytal** [comment]
Seems fixed on c.nc, closing

**Comment by AngryWarrior** [comment]
I have the exact same annoying issue with one file which is a backup of a wordpress .wp-config file.
![Image](https://github.com/user-attachments/assets/1cf66039-fba3-4401-ac2c-7df6e6c5ec94)

![Image](https://github.com/user-attachments/assets/70608135-6f5c-44c3-9fe6-a64cf8a1cc5b)

I have checked the server file permissions and they check out. Nothing incorrect in the file setting. 
I tired to go directly into the data folder were Nextcloud stores the file and delete/rename/ move. 

**None of the actions works.** 
Nextcloud refuses to do anything and just display the error message.

**Comment by come-nc** [comment]
@AngryWarrior Maybe look into `occ info:file` and `occ files:delete` and see if that helps or brings information.

**Comment by AllirWasTaken** [comment]
Same issue, every time i try to delete files that appear when you create new account some of them is not possible to be deleted or touched/renamed

**Comment by corpix** [comment]
For those who will come here from google (because this bug is still a thing in `31.0.2`):

> in my case deletion errors appeared after power loss while uploads was in progress.
> I opened directory in web interface, copied id from browser address bar, then connected to server using SSH, and:

```
sudo -u nextcloud nextcloud-occ files:delete --force -- 817806
...
"files/85aae0d472964297b11964af7058dcf3" is locked, existing lock on file: 9 shared locks 
...
```

It doesn't worked even with `--force`. Here is what I done:

- found directory on server using information obtained with `sudo -u nextcloud nextcloud-occ info:file 817806`
- removed it from terminal
- run full rescan for user with `sudo -u nextcloud nextcloud-occ files:scan -- corpix`

Rescan process reported 9 files removed.

**Comment by frostified-io** [comment]
I just had an issue that sounds very similar to the one described in this thread.

Two files were undeletable / unmovable with "action failed" in the Web GUI. It looked like it was just a corrupted pointer to files long gone. Interestingly those were duplicated when I made an entire copy of the folder.

I got rid of them by moving all files except those two entries into a newly created folder. Then I deleted the folder which contained these zombie file pointers. Now it looks good. Desktop client finally gives the "All Synced" again.
